src/wallet/test/coinselector_tests.cpp

```diff
@@ -449,6 +453,37 @@ BOOST_AUTO_TEST_CASE(bnb_search_test)
     }
 }
 
+BOOST_AUTO_TEST_CASE(tx_creation_bnb_sffo_restriction)
+{
+    // Verify the transaction creation process does not produce a BnB solution when SFFO is enabled.
+    // This is currently problematic because it could require a change output. And BnB is specialized on changeless solutions.
+    std::unique_ptr<CWallet> wallet = NewWallet(m_node);
+    WITH_LOCK(wallet->cs_wallet, wallet->SetLastBlockProcessed(300, uint256{})); // set a high block so internal UTXOs are selectable
+
+    FastRandomContext rand{};
+    CoinSelectionParams params{
+            rand,
+            /*change_output_size=*/ 31,
+            /*change_spend_size=*/ 68,
```

Comment:
in https://github.com/bitcoin/bitcoin/pull/28994/commits/06cd7e0bea4408cc508ad4074a44ddced3c773f8:

Why 31 and 68? It's a bit confusing as it seems like these numbers were chosen with intention, but when I changed them both to 0 the test still passed (and also still failed as expected with `params.m_subtract_fee_outputs = false`).

If there is a reason for choosing these numbers, it's worth documenting the reason in a comment. If the numbers are arbitrary, it would be better to use 0 for both as it better communicates that the choice of numbers is irrelevant to the test.

Reply:
31 -> p2wpkh output size.
68 -> p2wpkh input size (high-r signature).

The previous test version didn't have any hardcoded value. However, since we are now calling a low-level function, we need to hardcode them. Setting them to 0 would be incorrect. Even though they are not currently in use on this process, these values must align with those used in the wallet. We learned this lesson the hard way; the `coinselector_tests` is full of invalid values that lack real meaning, affecting the introduction of new features and other changes in the past. This is why #28985 is what it is.

Reply:
Cool, agree with using real-world values, but I think it's worth documenting why those specific values were chosen in a comment since the choice of values isn't relevant to the test.

Reply:
Pushed it. Thanks josi.